crates/relayer-framework/src/base/runtime/traits/runtime.rs

```diff
@@ -1,7 +1,9 @@
-use crate::base::core::traits::error::HasError;
+use crate::base::core::traits::error::HasErrorType;
 
-pub trait HasRuntime: HasError {
-    type Runtime: HasError<Error = Self::Error>;
+pub trait HasRuntime: HasErrorType {
+    type Runtime: HasErrorType;
 
     fn runtime(&self) -> &Self::Runtime;
+
+    fn runtime_error(e: <Self::Runtime as HasErrorType>::Error) -> Self::Error;
```

Comment:
I originally wanted `HasRuntime` to has `InjectError<<Self::Runtime as HasError>::Error>` as the supertrait. But I realized that it would require the implementor to have a generic implementation of `InjectError`, which is not allowed by Rust due to overlapping implementations.

So instead we would have explicit conversion methods for the sub-error types.